wagtail/wagtailadmin/edit_handlers.py

```diff
@@ -621,6 +621,16 @@ def required_formsets(cls):
             }
         }
 
+    @classmethod
+    def html_declarations(cls):
+        # We need to include html_declarations for StreamFields in the related model
+        declarations = [
+            field.stream_block.all_html_declarations()
+            for field in cls.related.related_model._meta.fields
+            if hasattr(field, 'stream_block')
+        ]
```

Comment:
Looks like all we really need here is `return cls.get_child_edit_handler_class().html_declarations()`. Packaging this up with some extra tests now...